src/test/regress/expected/rpt.out

```diff
@@ -1608,6 +1658,182 @@ drop table t1;
+explain (costs off, verbose) select * from t1 where a in (select random() from t where i=a group by i);
+                              QUERY PLAN                               
+-----------------------------------------------------------------------
+ Gather Motion 3:1  (slice1; segments: 3)
+   Output: t1.a
+   ->  Seq Scan on rpt.t1
+         Output: t1.a
+         Filter: (SubPlan 1)
+         SubPlan 1
+           ->  Result
+                 Output: random(), t.i
+                 Filter: (t.i = t1.a)
+                 ->  Materialize
+                       Output: t.i
+                       ->  Broadcast Motion 1:3  (slice2; segments: 1)
+                             Output: t.i
+                             ->  Seq Scan on rpt.t
+                                   Output: t.i
+ Optimizer: Postgres-based planner
+ Settings: enable_bitmapscan = 'off', enable_seqscan = 'off'
+(17 rows)
+
+explain (costs off, verbose) select * from t1 where a in (select random() from t where i=a);
+                              QUERY PLAN                               
+-----------------------------------------------------------------------
+ Gather Motion 3:1  (slice1; segments: 3)
+   Output: t1.a
+   ->  Seq Scan on rpt.t1
+         Output: t1.a
+         Filter: (SubPlan 1)
+         SubPlan 1
+           ->  Result
+                 Output: random()
+                 Filter: (t.i = t1.a)
+                 ->  Materialize
+                       Output: t.i
+                       ->  Broadcast Motion 1:3  (slice2; segments: 1)
+                             Output: t.i
+                             ->  Seq Scan on rpt.t
+                                   Output: t.i
+ Optimizer: Postgres-based planner
+ Settings: enable_bitmapscan = 'off', enable_seqscan = 'off'
+(17 rows)
+
+-- ensure we make broadcast motion when volatile function in deleting motion flow
+explain (costs off, verbose) insert into t2 (a, b) select i, random() from t;
+                         QUERY PLAN                          
+-------------------------------------------------------------
+ Insert on rpt.t2
+   ->  Broadcast Motion 1:3  (slice1; segments: 1)
+         Output: t.i, (random())
+         ->  Result
+               Output: t.i, (random())
+               ->  Seq Scan on rpt.t
+                     Output: t.i, random()
+ Optimizer: Postgres-based planner
+ Settings: enable_bitmapscan = 'off', enable_seqscan = 'off'
+(9 rows)
+
+-- ensure we make broadcast motion when volatile function in correlated subplan qual
+explain (costs off, verbose) select * from t1 where a in (select f(i) from t where i=a and f(i) > 0);
+                              QUERY PLAN                               
+-----------------------------------------------------------------------
+ Gather Motion 3:1  (slice1; segments: 3)
+   Output: t1.a
+   ->  Seq Scan on rpt.t1
+         Output: t1.a
+         Filter: (SubPlan 1)
+         SubPlan 1
+           ->  Result
+                 Output: f(t.i)
```

Comment:
This plan seems to be still invalid, because the executor may calculate different value of volatile function on each segment for distribution, declared as Replicated.

Reply:
I checked different queries on GP and Postgres and got the same resullts on each BD.
For example one of such queries.
GP:
```
postgres=# explain (costs off, verbose, analyze) select * from t1 where a in (select f(i) from t where i=a and f(i) > 0 group by i order by i limit 100);
                                                          QUERY PLAN                                                          
------------------------------------------------------------------------------------------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3) (actual time=1.687..1.756 rows=10 loops=1)
   Output: t1.a
   ->  Seq Scan on public.t1 (actual time=1.222..1.316 rows=5 loops=1)
         Output: t1.a
         Filter: (SubPlan 1)
         SubPlan 1
           ->  Limit (actual time=0.040..0.056 rows=1 loops=5)
                 Output: (f(t.i)), t.i
                 ->  GroupAggregate (actual time=0.040..0.055 rows=1 loops=5)
                       Output: f(t.i), t.i
                       Group Key: t.i
                       ->  Result (actual time=0.001..0.005 rows=1 loops=5)
                             Output: t.i
                             Filter: (t.i = t1.a)
                             ->  Materialize (actual time=0.162..0.163 rows=8 loops=6)
                                   Output: t.i
                                   work_mem: 50kB  Segments: 3  Max: 17kB (segment 0)  Workfile: (0 spilling)
                                   ->  Broadcast Motion 1:3  (slice2; segments: 1) (actual time=0.961..0.964 rows=10 loops=1)
                                         Output: t.i
                                         ->  Seq Scan on public.t (actual time=0.245..0.304 rows=10 loops=1)
                                               Output: t.i
                                               Filter: (f(t.i) > 0)
                                               Rows Removed by Filter: 1
 Optimizer: Postgres-based planner
 Planning Time: 2.474 ms
   (slice0)    Executor memory: 57K bytes.
   (slice1)    Executor memory: 77K bytes avg x 3 workers, 77K bytes max (seg0).  Work_mem: 17K bytes max.
   (slice2)    Executor memory: 46K bytes (seg2).
 Memory used:  128000kB
 Execution Time: 5.760 ms
(30 rows)

postgres=# select * from t1 where a in (select f(i) from t where i=a and f(i) > 0 group by i order by i limit 100);
 a  
----
  1
  5
  6
  9
 10
  2
  3
  4
  7
  8
(10 rows)
```
Postgres:
```
test=# explain (costs off, verbose, analyze) select * from t1 where a in (select f(i) from t where i=a and f(i) > 0 group by i order by f(i) limit 100);
                                        QUERY PLAN                                         
-------------------------------------------------------------------------------------------
 Seq Scan on public.t1 (actual time=61.550..61.595 rows=10 loops=1)
   Output: t1.a
   Filter: (SubPlan 1)
   Rows Removed by Filter: 1
   SubPlan 1
     ->  Limit (actual time=0.012..0.012 rows=1 loops=11)
           Output: (f(t.i)), t.i
           ->  Sort (actual time=0.011..0.011 rows=1 loops=11)
                 Output: (f(t.i)), t.i
                 Sort Key: (f(t.i))
                 Sort Method: quicksort  Memory: 25kB
                 ->  Group (actual time=0.010..0.010 rows=1 loops=11)
                       Output: f(t.i), t.i
                       Group Key: t.i
                       ->  Seq Scan on public.t (actual time=0.007..0.007 rows=1 loops=11)
                             Output: t.i
                             Filter: ((t.i = t1.a) AND (f(t.i) > 0))
                             Rows Removed by Filter: 10
 Query Identifier: 2321456564250273903
 Planning Time: 0.187 ms
 Execution Time: 63.126 ms
(21 rows)

select * from t1 where a in (select f(i) from t where i=a and f(i) > 0 group by i order by f(i) limit 100);
 a  
----
  1
  2
  3
  4
  5
  6
  7
  8
  9
 10
(10 rows)
```